doc/source/basics.rst

```diff
@@ -1726,6 +1726,24 @@ then the more *general* one will be used as the result of the operation.
    # conversion of dtypes
    df3.astype('float32').dtypes
 
+When trying to convert a subset of columns to a specified type using :meth:`~DataFrame.astype`  and :meth:`~DataFrame.loc`, utilizing **:** as mask, upcasting occurs.
+:meth:`~DataFrame.loc` tries to fit in what we are assigning to the current dtypes, while [ ] will overwrite them taking the dtype from the right hand side.
+
+.. ipython:: python
+
+   df = pd.DataFrame({'a': [1,2,3], 'b': [4,5,6], 'c': [7, 8, 9]})
+   df.loc[:, ['a', 'b']].astype(np.uint8).dtypes
+   df.loc[:, ['a', 'b']] = df.loc[:, ['a', 'b']].astype(np.uint8)
+
+To avoid this please take the following approach.
+
+.. ipython:: python
+
```

Comment:
flip these around showing the correct method of assignment using astype.

then you can put a ``::note`` for the explanation. People reading this just want the answer; *some* want the explanation, so highliting that differently helps.